packages/dashmate/src/listr/tasks/setup/regular/configureSSLCertificateTaskFactory.js

```diff
@@ -0,0 +1,142 @@
+              {
+                name: 'chainFilePath',
+                message: 'Path to certificate chain file',
+                validate: validateFileExists,
+              },
+              {
+                name: 'privateFilePath',
+                message: 'Path to certificate key file',
+                validate: validateFileExists,
+              },
+            ],
+            validate: ({ chainFilePath, privateFilePath }) => () => {
+              if (!validateFileExists(chainFilePath) || !validateFileExists(privateFilePath)) {
+                return false;
+              }
+
+              if (chainFilePath === privateFilePath) {
+                return 'the same path for both files';
+              }
+
+              return true;
+            },
+          });
+
+          ctx.certificate = fs.readFileSync(form.chainFilePath, 'utf8');
+          ctx.keyPair = {
+            privateKey: fs.readFileSync(form.privateFilePath, 'utf8'),
+          };
+
+          return saveCertificateTask(ctx.config);
+        },
+      },
+      [SSL_PROVIDERS.ZEROSSL]: {
+        title: 'Obtain ZeroSSL certificate',
+        task: async (ctx, task) => {
+          const apiKey = await task.prompt({
+            type: 'input',
+            message: 'Enter ZeroSSL API key',
+            validate: async (state) => {
+              try {
+                await listCertificates(state);
+
+                return true;
+              } catch (e) {
+                // do nothing
+              }
+
+              return 'Please enter a valid ZeroSSL API key';
+            },
+          });
+
+          ctx.config.set('platform.dapi.envoy.ssl.providerConfigs.zerossl.apiKey', apiKey);
+
+          return obtainZeroSSLCertificateTask(ctx.config);
+        },
+      },
+    };
+
+    return new Listr([
+      {
+        title: 'Configure SSL certificate',
+        task: async (ctx, task) => {
+          // TODO Move to upper function to show output in upper task?
+          const choices = [
+            { name: SSL_PROVIDERS.ZEROSSL, message: 'ZeroSSL' },
+            { name: SSL_PROVIDERS.FILE, message: 'File on disk' },
+          ];
+
+          if (ctx.preset !== PRESET_MAINNET) {
+            choices.push({ name: SSL_PROVIDERS.SELF_SIGNED, message: 'Self-signed' });
+          }
+
+          ctx.certificateProvider = await task.prompt({
+            type: 'select',
+            header: `  High-performance masternodes are required use TLS encryption on the DAPI
+  endpoint through which they service the network. This encryption is achieved
+  by loading an SSL certificate signed against the IP address specified in the
+  registration transaction. The certificate should be recognized by common web
+  browsers, and must therefore be issued by a well-known Certificate Authority
+  (CA). Dashmate offers three options to configure this certificate:
+
+    ZeroSSL      - Provide a (free) ZeroSSL API key and let dashmate configure the certificate
```

Comment:
Not sure, but it might be nice to include a URL to the relevant ZeroSSL page for getting an API key.

Reply:
Added

Reply:
The main link is okay. Is there a more specific page that would guide them to getting the API key? Maybe https://zerossl.com/documentation/api/  ("Access key" section)?